### PR TITLE
Add serialization tests

### PR DESCRIPTION
**Pull request by RemeshevskiyValeriy**
CON-314

### CODE REVIEW

File: tests/detached_editing/test_serialization.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    NgConnectTestCase,
+)
+
+
+class TestSerialization(NgConnectTestCase):
+    def setUp(self):
+        super().setUp()
+
+        self.types = [
+            (12345, "12345", 12345),  # integer
```
Comment (ivanbarsukov) [inline tests/detached_editing/test_serialization.py:24]: Please create a dataclass for this datasets. Without names it is difficult to understand the meaning without reading the tests code

File: tests/detached_editing/test_serialization.py [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+            (8388607, "8388607", 8388607),  # mediumint
+            (3.14159, "3.14159", 3.14159),  # real
+            ("Hello, World!", '"Hello, World!"', "Hello, World!"),  # text
```
Comment (ivanbarsukov) [inline tests/detached_editing/test_serialization.py:27]: It is also important to test for empty lines

File: tests/detached_editing/test_serialization.py [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+                QDate(2025, 3, 4),
+                json.dumps("2025-03-04"),
+                "2025-03-04",
+            ),  # date
+            (QTime(14, 30, 15), json.dumps("14:30:15"), "14:30:15"),  # time
+            (
+                QDateTime(QDate(2025, 3, 4), QTime(14, 30, 15)),
+                json.dumps("2025-03-04T14:30:15"),
+                "2025-03-04T14:30:15",
+            ),  # datetime
```
Comment (ivanbarsukov) [inline tests/detached_editing/test_serialization.py:38]: Invalid date, time and datetime should also be tested

File: tests/detached_editing/test_serialization.py [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+        ]
```
Comment (ivanbarsukov) [inline tests/detached_editing/test_serialization.py:39]: Need to test for None

File: tests/detached_editing/test_serialization.py [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+
+        self.geometries = [
+            "Point (1 1)",
+            "LineString (0 0, 1 1, 2 2)",
+            "Polygon ((0 0, 1 0, 1 1, 0 1, 0 0))",
+            "MultiPoint ((1 1),(2 2))",
+            "MultiLineString ((0 0, 1 1),(2 2, 3 3))",
+            "MultiPolygon (((0 0, 1 0, 1 1, 0 1, 0 0)),((2 2, 3 2, 3 3, 2 3, 2 2)))",
```
Comment (ivanbarsukov) [inline tests/detached_editing/test_serialization.py:47]: It is also important to test the geometry with the Z coordinate. The WKT format that NGW accepts is different from what QGIS API returns

File: tests/detached_editing/test_serialization.py [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+            (12345, "12345", 12345),  # integer
+            (8388607, "8388607", 8388607),  # mediumint
+            (3.14159, "3.14159", 3.14159),  # real
+            ("Hello, World!", '"Hello, World!"', "Hello, World!"),  # text
+            (
+                QDate(2025, 3, 4),
+                json.dumps("2025-03-04"),
+                "2025-03-04",
+            ),  # date
+            (QTime(14, 30, 15), json.dumps("14:30:15"), "14:30:15"),  # time
```
Comment (ivanbarsukov) [inline tests/detached_editing/test_serialization.py:33]: Please also add tests for types from the datetime library. It will require some code modifications.

File: tests/detached_editing/test_serialization.py [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+from tests.ng_connect_testcase import (
+    NgConnectTestCase,
+)
+
+
+class TestSerialization(NgConnectTestCase):
+    def setUp(self):
+        super().setUp()
+
+        self.types = [
```
Comment (ivanbarsukov) [inline tests/detached_editing/test_serialization.py:23]: It seems that `types` is not the word that reflects the meaning

File: tests/detached_editing/test_serialization.py [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+
+        self.geometries = [
+            "Point (1 1)",
+            "LineString (0 0, 1 1, 2 2)",
+            "Polygon ((0 0, 1 0, 1 1, 0 1, 0 0))",
+            "MultiPoint ((1 1),(2 2))",
+            "MultiLineString ((0 0, 1 1),(2 2, 3 3))",
+            "MultiPolygon (((0 0, 1 0, 1 1, 0 1, 0 0)),((2 2, 3 2, 3 3, 2 3, 2 2)))",
+        ]
+
```
Comment (ivanbarsukov) [inline tests/detached_editing/test_serialization.py:49]: Please also add tests for simplify_* methods. For dates, it will be important to check that both simplify methods return the same values.

**Review by ivanbarsukov — CHANGES_REQUESTED** [review]

**Review by ivanbarsukov — APPROVED** [review]
